src/libcore/intrinsics.rs

```diff
@@ -414,101 +414,198 @@ extern "rust-intrinsic" {
 
     /// Returns the square root of an `f32`
+    #[cfg(stage0)]
     pub fn sqrtf32(x: f32) -> f32;
     /// Returns the square root of an `f64`
+    #[cfg(stage0)]
     pub fn sqrtf64(x: f64) -> f64;
+    /// Returns the square root of a floating point type `T`.
+    #[cfg(not(stage0))]
+    pub fn sqrt<T>(x: T) -> T;
 
     /// Raises an `f32` to an integer power.
+    #[cfg(stage0)]
     pub fn powif32(a: f32, x: i32) -> f32;
     /// Raises an `f64` to an integer power.
+    #[cfg(stage0)]
     pub fn powif64(a: f64, x: i32) -> f64;
+    /// Raises a floating point to an integer power.
+    #[cfg(not(stage0))]
+    pub fn powi<T>(a: T, x: i32) -> T;
 
     /// Returns the sine of an `f32`.
+    #[cfg(stage0)]
     pub fn sinf32(x: f32) -> f32;
     /// Returns the sine of an `f64`.
+    #[cfg(stage0)]
     pub fn sinf64(x: f64) -> f64;
+    /// Returns the sine of a floating point type `T`.
+    #[cfg(not(stage0))]
+    pub fn sin<T>(x: T) -> T;
 
     /// Returns the cosine of an `f32`.
+    #[cfg(stage0)]
     pub fn cosf32(x: f32) -> f32;
     /// Returns the cosine of an `f64`.
+    #[cfg(stage0)]
     pub fn cosf64(x: f64) -> f64;
+    /// Returns the cosine of a floating point type `T`.
+    #[cfg(not(stage0))]
+    pub fn cos<T>(x: T) -> T;
 
     /// Raises an `f32` to an `f32` power.
+    #[cfg(stage0)]
     pub fn powf32(a: f32, x: f32) -> f32;
     /// Raises an `f64` to an `f64` power.
+    #[cfg(stage0)]
     pub fn powf64(a: f64, x: f64) -> f64;
+    /// Raises floating point of type `T` to a floating point of type `T` power.
+    #[cfg(not(stage0))]
+    pub fn pow<T>(a: T, x: T) -> T;
 
     /// Returns the exponential of an `f32`.
+    #[cfg(stage0)]
     pub fn expf32(x: f32) -> f32;
     /// Returns the exponential of an `f64`.
+    #[cfg(stage0)]
     pub fn expf64(x: f64) -> f64;
+    /// Returns the exponential of a foating point of type `T`.
+    #[cfg(not(stage0))]
+    pub fn exp<T>(x: T) -> T;
 
     /// Returns 2 raised to the power of an `f32`.
+    #[cfg(stage0)]
     pub fn exp2f32(x: f32) -> f32;
     /// Returns 2 raised to the power of an `f64`.
+    #[cfg(stage0)]
     pub fn exp2f64(x: f64) -> f64;
+    /// Returns 2 raised to the power of a floating point of type `T`.
+    #[cfg(not(stage0))]
+    pub fn exp2<T>(x: T) -> T;
 
     /// Returns the natural logarithm of an `f32`.
+    #[cfg(stage0)]
     pub fn logf32(x: f32) -> f32;
     /// Returns the natural logarithm of an `f64`.
+    #[cfg(stage0)]
     pub fn logf64(x: f64) -> f64;
+    /// Returns the natural logarithm of a floating point of type `T`.
+    #[cfg(not(stage0))]
+    pub fn log<T>(x: T) -> T;
 
     /// Returns the base 10 logarithm of an `f32`.
+    #[cfg(stage0)]
     pub fn log10f32(x: f32) -> f32;
     /// Returns the base 10 logarithm of an `f64`.
+    #[cfg(stage0)]
     pub fn log10f64(x: f64) -> f64;
+    /// Returns the base 10 logarithm of a foating point of type `T`.
+    #[cfg(not(stage0))]
+    pub fn log10<T>(x: T) -> T;
```

Comment:
To cut down on the annotation burden here, could you extract all the intrinsics to their own `extern "rust-intrinsic"` blocks? That way the `cfg` annotations should only be needed once.